odoo/addons/base/res/res_currency.py

```diff
@@ -43,11 +44,7 @@ class Currency(models.Model):
         ('rounding_gt_zero', 'CHECK (rounding>0)', 'The rounding factor must be greater than 0!')
     ]
 
-    @api.multi
-    def _compute_current_rate(self):
-        date = self._context.get('date') or fields.Date.today()
-        company_id = self._context.get('company_id') or self.env['res.users']._get_company().id
-        # the subquery selects the last rate before 'date' for the given currency/company
+    def get_rates(self, company_id, date):
```

Comment:
Maybe typecheck `company_id` and `date` just in case, to avoid the risk of people swapping them?

& add docstring?

Reply:
suggestion: change `company_id` (record id) to `company` (record)